digitaltwins/samples/device/javascript/pnpSampleDevice.js

```diff
@@ -0,0 +1,194 @@
+    break;
+  }
+};
+
+// TODO Not sure if this needs to be async or not coz we probably dont care if the response rached Hub.
+const sendCommandResponse = async (request, response, status, payload) => {
+  helperLogCommandRequest(request);
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  };
+};
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName, actualResponse) => {
+  const componentPart = componentPrefix + componentName;
+  const patch = { [componentPart]: {} };
+
+  Object.keys(propertiesToReport).forEach((propertyName) => {
+    const propertyContent = { value: propertiesToReport[propertyName] };
+    if (actualResponse) {
+      propertyContent.ac = actualResponse.code;
+      propertyContent.ad = actualResponse.description;
+      propertyContent.av = actualResponse.version;
+    }
+    patch[componentPart][propertyName] = propertyContent;
+  });
+  console.log('The following properties will be updated for component:' + componentName);
+  console.log(patch);
+  return patch;
+};
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log('Properties have been reported for component:' + componentName);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin, componentName) => {
+  const prefixAndComponentName = componentPrefix + componentName;
+  const versionProperty = '$version';
+
+  deviceTwin.on('properties.desired.' + prefixAndComponentName, (delta) => {
+    Object.entries(delta).forEach(([propertyName, propertyValue]) => {
+      propertyUpdateHandler(deviceTwin, componentName, propertyName, null, propertyValue.value, deviceTwin.properties.desired[versionProperty]);
+    });
+  });
+};
+
+
+const sendTelemetryAtInterval = async (deviceClient, componentName, secs) => {
+  let index = 0;
+  const interval = parseFloat(secs)*1000;
+  setInterval( async () => {
+    console.log('Sending telemetry message %d...', index);
+    const data = JSON.stringify({ temp: 1 + (Math.random() * 90), humidity: 1 + (Math.random() * 99) });
+    const pnpMsg = new Message(data);
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+    pnpMsg.contentType = 'application/json';
+    pnpMsg.contentEncoding = 'utf-8';
+    await deviceClient.sendEvent(pnpMsg);
+    index += 1;
+  }, interval);
+};
```

Comment:
const sendTelemetryAtInterval = async (deviceClient, componentName, secs) => {
  const interval = parseFloat(secs)*1000;
  let index = 0;
  while (true) {
    console.log('Sending telemetry message %d...', index++);
    const data = JSON.stringify({ temp: 1 + (Math.random() * 90), humidity: 1 + (Math.random() * 99) });
    const pnpMsg = new Message(data);
    pnpMsg.properties.add(messageSubjectProperty, componentName);
    pnpMsg.contentType = 'application/json';
    pnpMsg.contentEncoding = 'utf-8';
    await deviceClient.sendEvent(pnpMsg);
    await sleep(interval);
  }
};

Reply:
i think the reason this will not work (unless we change the telemetry to be last thing that we do) is because we will keep on scheduling awaits and these will all be scheduled in the microtask queue which has a higher preference of getting processed immediately...so we will never be able to schedule the other things as we are in a while true loop...plz correct me if i am mistaken